src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,269 @@
+#include "widget/wfindonwebmenu.h"
+
+#include <QDesktopServices>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/math.h"
+#include "util/parented_ptr.h"
+#include "util/qt.h"
+#include "util/widgethelper.h"
+
+namespace {
+
+QString composeActionText(const QString& prefix, const QString& trackProperty) {
+    return prefix + QStringLiteral(" | ") + trackProperty;
+}
+
+QString composeSearchQuery(const QString& trackAlbumOrTitle, const QString& artist) {
+    return trackAlbumOrTitle + QStringLiteral(" ") + artist;
+}
+
+QString composePrefixAction(WFindOnWebMenu::TrackSearchProperties trackSearchProperty) {
+    if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Title) {
+        return QObject::tr("Title");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Artist) {
+        return QObject::tr("Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+        return QObject::tr("Title + Artist");
+    } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndAlbum) {
+        return QObject::tr("Album + Artist");
+    } else {
+        return QObject::tr("Album");
+    }
+}
+
+const QString ksearchUrlSoundCloudArtist = QStringLiteral("https://soundcloud.com/search/people?");
```

Comment:
```suggestion
const QString kSearchUrlSoundCloudArtist = QStringLiteral("https://soundcloud.com/search/people?");
```
(Sorry for keeping nitpicking ...)

Reply:
It's totally okay. I have must missed that.